src/app/demo/demo.component.ts

```diff
@@ -27,7 +28,71 @@ export class DemoComponent implements OnInit {
 
   ngOnInit(): void {}
 
+  // click botton btn just once
+  clicked = false;
+
+  //addTranslationLine is called when the user clicks the "Add Translation" button
+  addTranslationLine(): void {
+    // @ts-ignore
+    const readalongRoot: any = document.querySelector("read-along").shadowRoot;
+    const sentences = readalongRoot.querySelectorAll(".sentence");
+    sentences.forEach((sentence: any) => {
+      const innerbutton = document.createElement("button");
+      innerbutton.innerHTML = "Button";
+      // innerbutton.setAttribute("click", "clicked = true");
+      innerbutton.addEventListener("click", () => {
+        sentence.insertAdjacentHTML(
+          "beforeend",
+          '<br><span class = "translation" contenteditable = True>Translation</span>'
+        );
+        innerbutton.disabled = true;
+      });
+      sentence.insertAdjacentElement("afterend", innerbutton);
+    });
+  }
+
+  //pass all sentences to b64Inputs[1]
+  updateTextXML(): void {
+    // @ts-ignore
+    const readalongRoot: any = document.querySelector("read-along").shadowRoot;
+    if (readalongRoot == null) {
+      return;
+    }
+    const translation = readalongRoot.querySelectorAll(".translation");
+    var textXML = this.b64Service.b64_to_utf8(
+      this.b64Inputs[1].substring(this.b64Inputs[1].indexOf(",") + 1)
+    );
+
+    const parser = new DOMParser();
+    const doc = parser.parseFromString(textXML, "application/xml");
+    // if translation class exist, delete it
+    if (doc.querySelector(".translation") != null) {
+      doc.querySelectorAll(".translation").forEach((node) => {
+        node.remove();
+      });
+    }
+    const ss = doc.querySelectorAll("s");
+    let count = 0;
+    ss.forEach((tag_s) => {
+      tag_s.insertAdjacentHTML(
+        "afterend",
+        `<span class="translation" contenteditable="true">${translation[count].innerHTML}</span>`
+      );
+      count++;
+    });
```

Comment:
It is also okay use span to insert the new translation line into the xml file and make html visible to that change. But xiaotong seems to add the similar changes about this function.

Reply:
Combine the milestone3 together of one brunch